File: Python/ejercicio_10.py
def main():
    enunciado = """
=========================================================
EJERCICIO 10
Sea X una variable aleatoria discreta con función de probabilidad:
P(X=1)=0.2, P(X=2)=0.3, P(X=3)=0.4, P(X=4)=0.1.
a) Calcule la media (esperanza) de X.
b) Calcule la varianza de X.
c) Calcule P(1 ≤ X ≤ 3).
=========================================================
"""
    print(enunciado)

    x_vals = [1, 2, 3, 4]
    p_x = [0.2, 0.3, 0.4, 0.1]
    
    print("--- SOLUCIÓN a) ---")
    print("Explicación: La media (o valor esperado) de una variable discreta se calcula sumando el producto de cada valor por su probabilidad: E[X] = sum(x * P(x)).")
    media = sum(x * p for x, p in zip(x_vals, p_x))
    print(f"-> Media E[X] = (1*0.2) + (2*0.3) + (3*0.4) + (4*0.1) = {media:.4f}\n")
    
    print("--- SOLUCIÓN b) ---")
    print("Explicación: La varianza se define como Var(X) = E[X^2] - (E[X])^2.")
    print("Primero calculamos la esperanza de X^2 multiplicando cada x al cuadrado por su probabilidad, y luego restamos el cuadrado de la media calculada en a).")
    esperanza_x2 = sum((x ** 2) * p for x, p in zip(x_vals, p_x))
    varianza = esperanza_x2 - (media ** 2)
    print(f"-> E[X^2] = (1^2*0.2) + (2^2*0.3) + (3^2*0.4) + (4^2*0.1) = {esperanza_x2:.4f}")
    print(f"-> Var(X) = {esperanza_x2:.4f} - ({media:.4f})^2 = {varianza:.4f}\n")
    
    print("--- SOLUCIÓN c) ---")
    print("Explicación: Para P(1 ≤ X ≤ 3), simplemente sumamos las probabilidades individuales de que X valga 1, 2 o 3.")
    p_1_a_3 = p_x[0] + p_x[1] + p_x[2]
    print(f"-> P(1 ≤ X ≤ 3) = P(X=1) + P(X=2) + P(X=3) = 0.2 + 0.3 + 0.4 = {p_1_a_3:.4f}")

if __name__ == "__main__":
    main()

File: R/ejercicio_10.R
enunciado <- "
=========================================================
EJERCICIO 10 (R)
Una variable aleatoria continua Y tiene función de densidad f(y) = (3/2)·y² para -1 ≤ y ≤ 1, y 0 en otro caso.
a) Verifique que es una densidad.
b) Calcule P(0 ≤ Y ≤ 0.5).
c) Obtenga la media y la varianza de Y.
=========================================================
"
cat(enunciado)

cat("--- SOLUCIÓN a) ---\n")
cat("Explicación: Verificamos matemáticamente que f(y) integra a 1 en todo su dominio [-1, 1]. Además f(y) es siempre positiva.\n")
cat("La integral analítica de (3/2)*y^2 es y^3 / 2.\n")
cat("Evaluada entre -1 y 1: (1^3 / 2) - ((-1)^3 / 2) = (1/2) - (-1/2) = 1. Sí es densidad válida.\n\n")

cat("--- SOLUCIÓN b) ---\n")
cat("Explicación: Evaluamos la misma integral (y^3 / 2) en el intervalo solicitado [0, 0.5].\n")
prob_rango <- (0.5^3) / 2 - (0^3) / 2
cat(sprintf("-> P(0 ≤ Y ≤ 0.5) = (0.5^3 / 2) - 0 = %.4f\n\n", prob_rango))

cat("--- SOLUCIÓN c) ---\n")
cat("Explicación: La media E[Y] se calcula con la integral de y*f(y) = (3/2)*y^3 de -1 a 1.\n")
cat("La primitiva es (3/8)*y^4. Por simetría (una función impar evaluada en un rango simétrico), el resultado es 0.\n")
media <- 0
cat(sprintf("-> Media E[Y] = %.4f\n\n", media))

cat("Explicación: La varianza es E[Y^2] - (E[Y])^2. Integramos y^2 * f(y) = (3/2)*y^4 de -1 a 1.\n")
cat("La primitiva es (3/10)*y^5. Evaluada entre -1 y 1 da: (3/10)*1 - (3/10)*(-1) = 6/10 = 0.6.\n")
var_y <- 0.6 - (media^2)
cat(sprintf("-> E[Y^2] = 0.6000\n"))
cat(sprintf("-> Varianza Var(Y) = 0.6000 - 0 = %.4f\n", var_y))
